prettyprinter/src/Data/Text/Prettyprint/Doc/Internal.hs

```diff
@@ -475,7 +475,7 @@ line' = FlatAlt Line mempty
 -- 'softline' = 'group' 'line'
 -- @
 softline :: Doc ann
-softline = group line
+softline = Union (Char ' ') Line
```

Comment:
Note that a "literal" inlining would have produced

```suggestion
softline = Union (Char ' ') (FlatLine Line (Char ' '))
```

I'm quite convinced that the version I went with is equivalent, but it would be nice to have some kind of "proof" for this…

Reply:
https://github.com/quchen/prettyprinter/issues/115 is related.

Reply:
This seems to be safe since the only code that cares about the right `FlatAlt` branch is in `changesUponFlattening`. However `changesUponFlattening` also ignores any right `Union` branch, which we're producing in this case.